Consolidating `w3up` and `upload-service`

alanshaw opened the pull request:
[📚 Preview](https://github.com/storacha/RFC/blob/rfc/consolidating-w3up-and-upload-service/rfc/consolidating-w3up-and-upload-service.md)

hannahhoward reviewed (commented):
This sounds involved enough that it should happen BEFORE we monorepo. Do you agree? (otherwise the merge could be complicated)

Am I correct that this would actually have the prod network working with the routing protocol? (just data would ONLY go to a single SP running CARPARK)

Overall makes sense if we can implement quickly. Is this something you can get help with or would it be just you? Maybe @fforbeck or @Peeja ?

alanshaw commented on rfc/consolidating-w3up-and-upload-service.md:
```diff
@@ -0,0 +1,72 @@
+
+* Change the `w3infra` service DID to `did:web:upload.storacha.network` and URL to `https://upload.storacha.network`.
+* Point `https://up.web3.storage` and `https://up.storacha.network` to `https://upload.storacha.network`.
+* Allow the service to receive invocations to audience `did:web:web.storage` and accept attestations signed by `did:web:web.storage`.
```
Woops! Sorry, yes typo.

alanshaw commented on rfc/consolidating-w3up-and-upload-service.md:
```diff
@@ -0,0 +1,72 @@
+* Add the legacy Store Protocol handlers to `upload-service` by importing existing `@web3-storage/upload-api`.
+* Setup and deploy a production storage provider node that writes to carpark.
+* Migrate `allocation` table data to `blob-registry` table - this allows `space/blob/list` capability to work for older spaces.
+* Build a maintenance mode switch to existing system allowing migration to take place without writes.
```
I've never used that but sounds sensible!

alanshaw commented:
@Peeja @fforbeck @hannahhoward FYI I've updated the migration strategy (write to both) and removed the task to create a maintenance switch. I'll merge this now and create an issue in the project planning with the task/people allocations we discussed.
